### PR TITLE
FIX: No longer depending on pydotplus (networkx >=2.0 update)

### PR DESCRIPTION
**Pull request by TheChymera**
@effigies

Fixes: https://github.com/nipy/nipype/issues/3244

didn't commit after running `make check-before-commit` because it changes a lot of files with what appear to be unrelated changes.
Also it fails with:

```
chymera@silenthost ~/src/nipype $ make check-before-commit
Checking specs and autogenerating spec tests
env PYTHONPATH=".:" python tools/checkspecs.py
CompCor has nonautotest
TSNR has nonautotest
200917-12:08:45,637 nipype.interface WARNING:
	 VTK was not found
200917-12:08:45,638 nipype.interface WARNING:
	 tvtk wasn't found
ErrorMap has nonautotest
Overlap has nonautotest
Overlap has nonautotest
TSNR has nonautotest
200917-12:08:52,951 nipype.interface INFO:
	 We advise you to upgrade DIPY version. This upgrade will open access to more function
200917-12:08:52,953 nipype.interface INFO:
	 We advise you to upgrade DIPY version. This upgrade will open access to more function
200917-12:08:52,955 nipype.interface INFO:
	 We advise you to upgrade DIPY version. This upgrade will open access to more models
FSSurfaceCommand has nonautotest
BBRegister has nonautotest
FILMGLS has nonautotest
Traceback (most recent call last):
  File "tools/checkspecs.py", line 522, in <module>
    ic.check_modules()
  File "tools/checkspecs.py", line 475, in check_modules
    bad_specs = self.test_specs(m)
  File "tools/checkspecs.py", line 230, in test_specs
    __import__(uri)
  File "/home/chymera/src/nipype/nipype/sphinxext/apidoc/__init__.py", line 5, in <module>
    from sphinxcontrib.napoleon import (
ModuleNotFoundError: No module named 'sphinxcontrib.napoleon'
make: *** [Makefile:73: specs] Error 1
```

I know I ran into this before, maybe this is another unnecessary dep best fixed in another PR.

## Acknowledgment

- [x] \(Mandatory\) I acknowledge that this contribution will be available under the Apache 2 license.

### CODE REVIEW

File: nipype/interfaces/cmtk/nx.py [hunk 1]
```diff
@@ -289,11 +289,6 @@ def compute_singlevalued_measures(ntwk, weighted=True, calculate_cliques=False):
     iflogger.info("Computing single valued measures:")
     measures = {}
     iflogger.info("...Computing degree assortativity (pearson number) ...")
-    try:
-        measures["degree_pearsonr"] = nx.degree_pearsonr(ntwk)
-    except AttributeError:  # For NetworkX 1.6
-        measures["degree_pearsonr"] = nx.degree_pearson_correlation_coefficient(ntwk)
```
Comment (effigies) [inline nipype/interfaces/cmtk/nx.py:295]: I think we want to keep this one. And below, we'll want to keep `degree_assortativity_coefficient`.

File: nipype/info.py [hunk 2]
```diff
@@ -99,7 +99,7 @@ def get_nipype_gitversion():
 
 # versions
 NIBABEL_MIN_VERSION = "2.1.0"
-NETWORKX_MIN_VERSION = "1.9"
+NETWORKX_MIN_VERSION = "2.0"
```
Comment (effigies) [inline nipype/info.py:102]: And remove L146 with `pydotplus`?

File: nipype/interfaces/cmtk/nx.py (hunk 1 above)
Comment (TheChymera) [reply nipype/interfaces/cmtk/nx.py:295]: why keep it if it's just for `=networkx-1.6`, when we are no longer supporting `<networkx-2.0`?

Comment (effigies) [reply nipype/interfaces/cmtk/nx.py:295]: I checked, and that's the newer of the two.

Comment (TheChymera) [reply nipype/interfaces/cmtk/nx.py:295]: @effigies sorry, I don't understand. 1.6 is greater than 2.0?

Comment (effigies) [reply nipype/interfaces/cmtk/nx.py:295]: No, the `degree_pearsonr` was the function name in `<1.6`. `degree_pearson_correlation_coefficient` is in `>=1.6`.

I agree it seems backwards. I usually put the deprecated version in the `except` block.

Comment (TheChymera) [reply nipype/interfaces/cmtk/nx.py:295]: > I agree it seems backwards. I usually put the deprecated version in the except block.

Ah, yes, that was what confused me. Thank you for checking!

File: nipype/info.py (hunk 2 above)
Comment (TheChymera) [reply nipype/info.py:102]: wait, so we are dropping both pydot and pydotplus?

**Comment by TheChymera** [comment]
@effigies ^ ?

**Comment by effigies** [comment]
Pushed a commit to `master` that I think should fix the test failures. If everything's green, feel free to rebase on that.

**Comment by effigies** [comment]
Test failures fixed on `master`.

**Comment by TheChymera** [comment]
@effigies rebased, but the circleci tests seem to still fail. Any ideas why? They look like permission errors, no idea how that could be related to the PR....

**Comment by effigies** [comment]
Not sure. My guess is there's something weird with running CircleCI on your own fork, rather than as a PR on nipy. But Travis is passing, so let's go ahead with this. If there's a continued problem with Circle, we can address it when it reproduces on nipy.

**Pull request merged**